src/btree/bt_read.c

```diff
@@ -125,17 +128,258 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+    for (; ret == 0; ret = cursor->next(cursor)) {
+        if (!first_scan) {
+            WT_ERR(
+              cursor->get_key(cursor, &las_btree_id, &next_las_key, &las_timestamp, &las_txnid));
+
+            /* Set the key to check with maximum timestamp and transaction id.*/
+            cursor->set_key(cursor, las_btree_id, &next_las_key, WT_TS_MAX, WT_TXN_MAX);
+        }
+
+        WT_ERR(cursor->search_near(cursor, &exact));
+        /*
+         * check whether the search_near returned the same key or not? In case if it returns next
+         * key, continue with the previous record.
+         */
+        if (exact > 0)
+            WT_ERR(cursor->prev(cursor));
+
+        WT_ERR(cursor->get_key(cursor, &las_btree_id, &las_key, &las_timestamp, &las_txnid));
+        first_scan = false;
+
+        /* Stop before crossing over to the next btree */
+        if (las_btree_id != S2BT(session)->id)
+            break;
+
+        WT_ERR(__wt_compare(session, NULL, &las_key, &ref->page_las->min_las_key, &cmp));
+        /* we got a key that is less than the minimum key of the page */
+        if (cmp < 0)
+            break;
+
+        WT_ERR(__wt_compare(session, NULL, &las_key, &ref->page_las->max_las_key, &cmp));
+        /* we reached the end of the key of the LAS records related to the current page */
+        if (cmp > 0)
+            break;
+
+        instantiated_cnt++;
+        if (ref->page_las->birthmarks_cnt > 0) {
+            for (birthmarkp = ref->page_las->birthmarks + birthmark_cnt;
+                 birthmark_cnt < ref->page_las->birthmarks_cnt; birthmark_cnt++, birthmarkp++) {
+                if (birthmarkp->instantiated)
+                    continue;
+
+                WT_ERR(__wt_compare(session, NULL, &las_key, &birthmarkp->key, &cmp));
+                if (cmp == 0) {
+                    WT_ERR(__create_birthmark_upd(session, birthmarkp, &upd));
+                    birthmark_record = true;
+                    break;
+                } else if (cmp < 0)
+                    /* LAS key is less than birthmark key, so no birthmark record for that key */
+                    break;
+                else
+                    /* LAS key is greater than birthmark key, check next set of birthmark records */
+                    continue;
+            }
+        }
+
+        if (upd == NULL) {
+            /* Allocate the WT_UPDATE structure. */
+            WT_ERR(
+              cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+
+            WT_ERR(__wt_update_alloc(session, &las_value, &upd, &incr, upd_type));
+            upd->txnid = las_txnid;
+            upd->durable_ts = durable_timestamp;
+            upd->start_ts = las_timestamp;
+            upd->prepare_state = prepare_state;
+        }
+
+        /* Ignore the next set of records for the same key that is already instantiated. */
```

Comment:
This comment doesn't quite make sense to me. Also I think it belongs near the code where you're skipping to the next key.

Reply:
This comment is no longer required. removed.